src/python-common/ceph/deployment/drive_selection/selector.py

```diff
@@ -68,7 +83,7 @@ def _limit_reached(device_filter, len_devices,
         """
         limit = device_filter.limit or 0
 
-        if limit > 0 and len_devices >= limit:
+        if limit > 0 and (len_devices >= limit or len(self.osds_for_spec) >= limit):
```

Comment:
can you move this into a different commit?

Reply:
done